From 8587f3ae7c190fc527887965f418d390e2c0226c Mon Sep 17 00:00:00 2001
From: Nolen Johnson <johnsonnolen@gmail.com>
Date: Sat, 14 Jul 2018 15:12:40 -0400
Subject: [PATCH 2/2] sepolicy: public: Exclude Recovery from system mount
 neverallow

* This is needed to allow recovery to (re)mount /system.

Change-Id: I92ea22c12872f734bddbb47672eb1e9f36e33ba0
---
 public/domain.te | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/public/domain.te b/public/domain.te
index c1a7827b..8d0b507b 100644
--- a/public/domain.te
+++ b/public/domain.te
@@ -410,7 +410,7 @@ neverallow { domain -recovery -kernel -update_engine with_asan(`-asan_extract')
 
 # Don't allow mounting on top of /system files or directories
 neverallow * exec_type:dir_file_class_set mounton;
-neverallow { domain -init } { system_file vendor_file_type }:dir_file_class_set mounton;
+neverallow { domain -init userdebug_or_eng(`-recovery') } { system_file vendor_file_type }:dir_file_class_set mounton;
 
 # Nothing should be writing to files in the rootfs.
 neverallow { domain userdebug_or_eng(`-recovery -update_engine') } rootfs:file { create write setattr relabelto append unlink link rename };
-- 
2.17.1

